velox/exec/OutputBufferManager.cpp

```diff
@@ -17,21 +17,14 @@
 #include "velox/exec/Task.h"
 
 namespace facebook::velox::exec {
-// static
-void OutputBufferManager::initialize(const Options& options) {
-  std::lock_guard<std::mutex> l(initMutex_);
-  VELOX_CHECK(
-      instance_ == nullptr, "May initialize OutputBufferManager only once");
-  instance_ = std::make_shared<OutputBufferManager>(options);
-}
 
 // static
-std::weak_ptr<OutputBufferManager> OutputBufferManager::getInstance() {
-  std::lock_guard<std::mutex> l(initMutex_);
-  if (!instance_) {
-    instance_ = std::make_shared<OutputBufferManager>(Options());
-  }
-  return instance_;
+std::weak_ptr<OutputBufferManager> OutputBufferManager::getInstance(
+    const Options* optionsPtr) {
+  auto options = optionsPtr ? *optionsPtr : Options();
```

Comment:
We lose some check to make sure no new option is passed in if the instance is already created.

On the other hand, it seems `OutputBufferManager::initialize` is not used anywhere.  Maybe just drop the option parameter altogether?

Reply:
Thanks @Yuhta I have updated the code to
```
std::shared_ptr<OutputBufferManager> OutputBufferManager::getInstance() {
  static std::shared_ptr<OutputBufferManager> instance =
      std::make_shared<OutputBufferManager>(Options());
  return instance;
}
```
Appreciate your review again!